src/optimality.ts

```diff
@@ -0,0 +1,112 @@
+import { type Matrix } from 'ml-matrix';
+
+import setDifference from './util/setDifference';
+
+type Optimality = (
```

Comment:
Ditto. These types should be in the function declaration, with jsdoc to document them.